### ISSUE TITLE
[BUG] Rancher 2.8.5 - Constantly Crashing/OOM'ing after upgrade to EKS v1.28.8

### ISSUE THREAD

**Issue by ndlanier** [rancher/rancher#46327]
**Rancher Server Setup**
- Rancher version:
- Installation option (Docker install/Helm Chart): Helm
   - If Helm Chart, Kubernetes Cluster and version (RKE1, RKE2, k3s, EKS, etc): EKS 1.28.8
- Proxy/Cert Details:

**Information about the Cluster**
- Kubernetes version: 1.28.8
- Cloud provider: AWS

**User Information**
- What is the role of the user logged in? (Admin/Cluster Owner/Cluster Member/Project Owner/Project Member/Custom): Admin

**Describe the bug**
Following upgrading to EKS 1.28.8, the rancher pods began to crashloop from OOM'ing. Deployment was at 5 replicas, but have scaled down that down to 3. Bumped memory limit to 90G. They operate normally, then they queue up a lot of work that spikes to the memory limit and the pods crash. 

Rancher was previously at rancher 2.7.10, upgraded a little over a week ago (7/16/2024) to 2.8.5 with no issues.
EKS was previously at 1.27.12 upgraded today to 1.28.8

We have over 400 downstream clusters

**To Reproduce**

**Result**

**Expected Result**
Nominal startup/stable memory usage at start with no extreme memory spike

**Screenshots**

**Additional context**

**Comment by Rancheroo** [comment]
Great to hear the Rancher upgrade went well, just this Kubernetes upgrade that is causing issues. EKS is a product by AWS however, I'd like to gather some additional information.

Could you run the following command on the local Rancher cluster?

to view the Rancher logs
`kubectl logs -n cattle-system -l app=rancher -f -c rancher --tail=100`

to collect Rancher current and previous pod logs
`for pod in $(kubectl get pods -n cattle-system -l app=rancher --no-headers -o custom-columns=":metadata.name")
  do
    kubectl logs -n cattle-system $pod -c rancher |& gzip > $pod.log.gz
    kubectl logs -n cattle-system $pod -c rancher -p |& gzip > $pod.previous.log.gz
done
`

Please provide any relevant entries that are displayed in the output. Specifically, I'd like to investigate any entries with error messages or unusual activity that might be contributing to the memory spike. This will help us identify potential issues and narrow down the cause.

**Comment by moio** [comment]
It seems like this is a duplicate of https://github.com/rancher/rancher/issues/44347

**Comment by ndlanier** [comment]
Yup my team has been working with some SUSE engineers and #44347 is what caused the issue and the astronomical memory usage.

For us it was exacerbated by age of our rancher instance and downstream clusters, scale, and the usage of argocd talking to the rancher api.

**Comment by ndlanier** [comment]
Related to #43717 then as well

**Comment by ndlanier** [comment]
We deleted the duplicate secrets on whatever downstream clusters we could connect to, and we started seeing clusters get into good states again. We have since set up a cronjob to run daily to keep the impersonation secret count as low as we can.

We have about 3/4's of our downstream clusters recovered. 

We also disabled the cache on the management cluster, so that we could attempt to disable the cache on some downstream clusters but that didn't really get any to recover other than maybe 1.

For ones we could connect to via another tool, we were able to restart the kube-api-server and it would seem to restart the cluster agent deployment and that seemed to resolve a handful.
